geopandas/base.py

```diff
@@ -20,39 +22,60 @@ class RTreeError(Exception):
     HAS_SINDEX = False
 
 
-def _geo_op(this, other, op):
-    """Operation that returns a GeoSeries"""
-    if isinstance(other, GeoPandasBase):
-        this = this.geometry
-        crs = this.crs
-        if crs != other.crs:
-            warn('GeoSeries crs mismatch: {0} and {1}'.format(this.crs,
-                                                              other.crs))
-        this, other = this.align(other.geometry)
-        return gpd.GeoSeries([getattr(this_elem, op)(other_elem)
-                             for this_elem, other_elem in zip(this, other)],
-                             index=this.index, crs=crs)
+def binary_geo(op, left, right):
+    """Geometric operation that returns a pandas Series"""
+    if isinstance(right, GeoPandasBase):
+        left = left.geometry
+        left, right = left.align(right.geometry)
+        t = left._geometry_array
+        o = right._geometry_array
+        x = vectorized.vector_binary_geo(op, t.data, o.data)
+        return gpd.GeoSeries(x, index=left.index, crs=left.crs)
+    elif isinstance(right, BaseGeometry):
+        x = vectorized.binary_geo(op, left._geometry_array.data, right)
+        return gpd.GeoSeries(x, index=left.index, crs=left.crs)
     else:
-        return gpd.GeoSeries([getattr(s, op)(other)
-                             for s in this.geometry],
-                             index=this.index, crs=this.crs)
+        raise TypeError(type(left), type(right))
 
 
-# TODO: think about merging with _geo_op
-def _series_op(this, other, op, **kwargs):
+def binary_predicate(op, this, other, *args, **kwargs):
     """Geometric operation that returns a pandas Series"""
     null_val = False if op != 'distance' else np.nan
+    assert not kwargs
```

Comment:
we never use kwargs somewhere? (this is just an internal function, so we can check that) If not, I would just remove it?